core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -417,8 +417,8 @@ class half_t {
     // Use non-volatile val_ref to suppress:
     // "warning: implicit dereference will not access object of type ‘volatile
     // __half’ in statement"
-    auto val_ref = const_cast<impl_type&>(val);
-    val_ref      = __float2half(__half2float(const_cast<impl_type&>(val)) *
+    auto& val_ref = const_cast<impl_type&>(val);
```

Comment:
is this an rvaule ref now?
```suggestion
    impl_type& val_ref = const_cast<impl_type&>(val);
```

Reply:
Yes, exactly, but I prefer not to write the type twice.

Reply:
I think this was changed in #4156. If not, the comments there address why this warning is real, and this fix is not legitimate

Reply:
This was changed in #4156.